packages/webapp/src/components/Task/PureIrrigationTask/index.jsx

```diff
@@ -186,30 +186,33 @@ export default function PureIrrigationTask({
 
   return (
     <>
-      <Controller
-        control={control}
-        name={IRRIGATION_TYPE}
-        rules={{ required: true }}
-        render={({ field: { onChange, onBlur, onFocus } }) => {
-          return (
-            <ReactSelect
-              onFocus={onFocus}
-              label={t('ADD_TASK.IRRIGATION_VIEW.TYPE_OF_IRRIGATION')}
-              options={IrrigationTypeOptions}
-              onBlur={onBlur}
-              onChange={(e) => {
-                onChange(e);
-                setIrrigationTypeValue(e.value);
-                setValue(MEASUREMENT_TYPE, e.default_measuring_type);
-                setValue(IRRIGATION_TYPE_ID, e.irrigation_type_id);
-              }}
-              isDisabled={disabled}
-              value={IrrigationTypeOptions.find((options) => options.value === irrigation_type)}
-              defaultValue={getDefaultIrrigationTypeOptions}
-            />
-          );
-        }}
-      />
+      {IrrigationTypeOptions.length >= defaultIrrigationTaskTypes.length && (
```

Comment:
@wasswarichard 

Would this always be true? And I don't think we want to conditionally render this component do we?

Reply:
@Duncan-Brain we need to render the component after loading the irrigation types for the defaults to be displayed and it will always be true.

Reply:
I think the core problem is that you should be passing the data you want loaded from an existing container to pass in the data to the "pure" component. 

@JimmyRowland  I think tried to discuss this with Anant here: https://github.com/LiteFarmOrg/LiteFarm/pull/2336#discussion_r1036613780

For example: PurePestControl has data passed down through three other containers: TaskDetails, StepOne, and TaskReadOnly.